### PR TITLE
Fixing up Kubernetes Guestbook Examples

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #634

Also adds the ability to run the tests as part of our CI tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -95,30 +185,142 @@ jobs:
...
-      - name: Run ${{ matrix.tests-set }} Tests
-        run: make specific_test_set TestSet=${{ matrix.tests-set }}
-  lint:
+      - name: Create test infra
+        run: |
+          export STACK=${{ secrets.PULUMI_TEST_OWNER }}/ci-cluster-$(head /dev/urandom | LC_CTYPE=C tr -dc '[:lower:]' | head -c5)
+          echo "::set-env name=STACK::$STACK"
+          make setup_test_infra
+
+  test-infra-destroy:
```
Comment (lukehoban) [inline .github/workflows/ci.yml:194]: I'm a little confused overall in what's going on in this file now. Is this step really about destroying?  Why do we have all the steps copied twice?  Why does `kubernetes` need a separate block?

Can we keep all this simpler?

File: misc/scripts/destroy-ci-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -o nounset -o errexit -o pipefail
+
+echo Deleting ephemeral Kubernetes cluster...
+
+pushd misc/scripts/testinfra/
+pulumi stack select "${STACK}" && \
```
[the remark this answers is not in the document]
Comment (stack72) [reply misc/scripts/destroy-ci-cluster.sh:7]: this is a direct copy of the work to allow a test setup in the pulumi-kubernetes repo so we can definitely change this for the better :)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -32,11 +32,101 @@ env:
...
+#      - name: Install Yarn
+#        run: curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version ${{ matrix.yarn-version }}
+#      - name: Update PATH for Yarn
+#        run: |
+#          echo "::add-path::$HOME/.yarn/bin"
+#          echo "::add-path::$HOME/.config/yarn/global/node_modules/.bin"
+#      - name: setup linting tool
+#        run: make install
+#      - name: Lint typescript files
+#        run: make lint
```
[the remark this answers is not in the document]
Comment (stack72) [reply .github/workflows/ci.yml:124]: yes - sorry, this is a WIP and I just marked it as so

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stack72) [reply .github/workflows/ci.yml:194]: This *might* be able to get simpler - I might even break out Kubernetes to a different workflow file. The reason it's a separate block is that we want to be able to run AWS, GCP, Azure etc and at the same time, spin up our test infra for the Kubernetes tests

We can definitely do something to make this more obvious of course

**Comment by stack72** [comment]
I am going to suggest that this is a good first step! This will be refactored as part of our build and deploy changes - but right now, this at least gets some of the Kubernetes examples on the map

@lukehoban are you willing to let this be a good intermediate? :)

File: kubernetes-cs-guestbook/simple/Guestbook.cs [hunk 4]
```diff
@@ -256,7 +256,7 @@ public Guestbook()
         }
         else
         {
-            this.FrontendIp = frontendService.Status.Apply(status => status.LoadBalancer.Ingress[0].Hostname);
+            this.FrontendIp =frontendService.Status.Apply(status => status.LoadBalancer.Ingress[0].Ip ?? status.LoadBalancer.Ingress[0].Hostname);
```
Comment (lukehoban) [inline kubernetes-cs-guestbook/simple/Guestbook.cs:259]:
```suggestion
            this.FrontendIp = frontendService.Status.Apply(status => status.LoadBalancer.Ingress[0].Ip ?? status.LoadBalancer.Ingress[0].Hostname);
```

**Pull request merged**